client/webserver/api.go

```diff
@@ -1170,6 +1170,26 @@ func (s *WebServer) apiToggleRateSource(w http.ResponseWriter, r *http.Request)
 	writeJSON(w, simpleAck(), s.indent)
 }
 
+// apiDeleteArchiveRecords handles the '/deletearchivedrecords' API request.
+func (s *WebServer) apiDeleteArchivedRecords(w http.ResponseWriter, r *http.Request) {
+	form := new(deleteRecordsForm)
+	if !readPost(w, r, form) {
+		return
+	}
+
+	var olderThan time.Time
+	if form.OlderThanMs > 0 {
+		olderThan = time.UnixMilli(form.OlderThanMs)
+	}
+
+	err := s.core.DeleteArchivedRecordsWithBackup(&olderThan, form.SaveMatchesToFile, form.SaveOrdersToFile)
```

Comment:
When you don't set the time on the frontend, the payload is this: `{"olderThanMs":null,"saveMatchesToFile":false,"saveOrdersToFile":false}`

That's OK I suppose, but the issue is that in this handler code, we leave `olderThan` as the zero value for `time.Time`.  i.e. `0001-01-01 00:00:00 +0000 UTC`, which has a `UnixMilli()` value of `-62135596800000`.  Then in the `BoltDB` code, we do `olderThanB = uint64Bytes(uint64(olderThan.UnixMilli()))`, which first converts the negative `int64` to an `uint64` and then to bytes.  Not intended.

The docs for `(*Core) DeleteArchivedRecords` state: `If a nil *time.Time is provided, current time is used.`.  So, this `WebServer` handler should pass nil if `OlderThanMs` is not set.

So, let's use a `*time.Time` (pointer), and consider further if we want the form to be sumitted with `null` for `olderThanMs`.

Reply:
In addition to changing the http handler as described, a related change to the `BoltDB` code:

```diff
diff --git a/client/db/bolt/db.go b/client/db/bolt/db.go
index e23ad98b..6e8f9374 100644
--- a/client/db/bolt/db.go
+++ b/client/db/bolt/db.go
@@ -1867,7 +1867,7 @@ func (db *BoltDB) DeleteInactiveOrders(ctx context.Context, olderThan *time.Time
                finished   bool
                olderThanB []byte
        )
-       if olderThan != nil {
+       if olderThan != nil && !olderThan.IsZero() {
                olderThanB = uint64Bytes(uint64(olderThan.UnixMilli()))
        } else {
                olderThanB = uint64Bytes(timeNow())
@@ -2008,7 +2008,7 @@ func (db *BoltDB) DeleteInactiveMatches(ctx context.Context, olderThan *time.Tim
                finished   bool
                olderThanB []byte
        )
-       if olderThan != nil {
+       if olderThan != nil && !olderThan.IsZero() {
                olderThanB = uint64Bytes(uint64(olderThan.UnixMilli()))
        } else {
                olderThanB = uint64Bytes(timeNow())
```

Reply:
Oh, shoot. `var olderThan time.Time` was supposed to be a pointer `var olderThan *time.Time`